managed/services/agents/registry.go

```diff
@@ -197,12 +197,25 @@ func (r *Registry) register(stream agentpb.Agent_ConnectServer) (*pmmAgentInfo,
 	if err != nil {
 		return nil, err
 	}
+
 	var node *models.Node
 	err = r.db.InTransaction(func(tx *reform.TX) error {
-		node, err = authenticate(agentMD, tx.Querier)
+		var agent *models.Agent
+		node, agent, err = authenticate(agentMD, tx.Querier)
 		if err != nil {
 			return err
 		}
+
+		// port with default value or not changed - skipping
+		if agentMD.MetricsPort == 0 || (agent.ListenPort != nil && *agent.ListenPort == agentMD.MetricsPort) {
+			return nil
+		}
+
+		// registering new pmmAgent port in database for scrape config.
+		params := models.ChangeCommonAgentParams{Port: &agentMD.MetricsPort}
+		if _, err = models.ChangeAgent(tx.Querier, agentMD.ID, &params); err != nil {
+			return err
+		}
```

Comment:
let's make this update inside `authenticate` function somewhere here https://github.com/percona/pmm/pull/1237/files#diff-61640292d442a75c3b3b3aa4944a99aef9860ea2325e8c9bc86cbb00a627f6f0R292

Reply:
in that case you won't need to return `agent` and rename `agent` to `agentInfo` below.

Reply:
ok

Reply:
done